pkg/components/dashdiffs/formatter_json.go

```diff
@@ -161,23 +161,35 @@ func (f *JSONFormatter) Format(diff diff.Diff) (result string, err error) {
 func (f *JSONFormatter) formatObject(left map[string]interface{}, df diff.Diff) {
 	f.addLineWith(ChangeNil, "{")
 	f.push("ROOT", len(left), false)
-	f.processObject(left, df.Deltas())
+	if err := f.processObject(left, df.Deltas()); err != nil {
+		f.pop()
+		// TODO: Deal with error
+		return
+	}
+
 	f.pop()
 	f.addLineWith(ChangeNil, "}")
 }
 
 func (f *JSONFormatter) formatArray(left []interface{}, df diff.Diff) {
 	f.addLineWith(ChangeNil, "[")
 	f.push("ROOT", len(left), true)
-	f.processArray(left, df.Deltas())
+	if err := f.processArray(left, df.Deltas()); err != nil {
+		f.pop()
+		// TODO: Deal with error
```

Comment:
Return error, but not sure about the `f.Pop()` since returning this error should probably fail, but unsure.